src/Bicep.Local.Deploy/Extensibility/GrpcProxyLocalExtension.cs

```diff
@@ -0,0 +1,174 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Text.Json.Nodes;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Bicep.Local.Extension.Protocol;
+using Bicep.Local.Extension.Rpc;
+using Google.Protobuf.Collections;
+using Json.Pointer;
+using ExtensibilityV2 = Azure.Deployments.Extensibility.Core.V2.Models;
+using Rpc = Bicep.Local.Extension.Rpc;
+
+namespace Bicep.Local.Deploy.Extensibility
+{
+    public class GrpcProxyLocalExtension : ILocalExtension, IAsyncDisposable
```

Comment:
Same as the existing implementation, difference is that implements `ILocalExtension` and also that uses extensibility v2 data types.